pkg/distributor/distributor.go

```diff
@@ -448,6 +454,28 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	var validationErrors util.GroupedErrors
 	validationContext := d.validator.getValidationContextForTime(time.Now(), tenantID)
 
+	shardStreamsCfg := d.validator.Limits.ShardStreams(tenantID)
+	maybeShardByRate := func(stream logproto.Stream, pushSize int) {
+		if shardStreamsCfg.Enabled {
+			streams = append(streams, d.shardStream(stream, pushSize, tenantID)...)
+		} else {
+			streams = append(streams, KeyedStream{
+				HashKey: lokiring.TokenFor(tenantID, stream.Labels),
+				Stream:  stream,
+			})
+		}
```

Comment:
we can add a `return` to remove else block
```suggestion
		if shardStreamsCfg.Enabled {
			streams = append(streams, d.shardStream(stream, pushSize, tenantID)...)
			return
		}
		streams = append(streams, KeyedStream{
			HashKey: lokiring.TokenFor(tenantID, stream.Labels),
			Stream:  stream,
		})
	
```

Reply:
Resolved by https://github.com/grafana/loki/pull/14711/commits/ad1c0b2ad882121da20acda5e50db3833ac0a39d